packages/push/src/instruction/int/clamp.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    error::InstructionResult,
+    instruction::{Instruction, instruction_error::PushInstructionError},
+    push_vm::{HasStack, stack::PushOnto},
+};
+
+/// An instruction that clamps a value to within a given range.
+///
+/// # Inputs
+///
+/// The `Clamp` instruction takes the following inputs:
+///    - Stack of type `i64`
+///      - One value
+///      - One minimum value
+///      - One maximum value
+///
+/// # Behavior
+///
+/// The `Clamp` instruction takes three values from the top of the `i64` stack:
+/// `value`, `min`, and `max`.
+///
+/// If `value` is less than `min`, `min` replaces `value` on the stack.
+///
+/// If `value` is greater than `max`, `max` replaces `value` on the stack.
+///
+/// If `value` is within the range `[min, max]` (inclusive on both sides), it
+/// remains unchanged.
+///
+/// ## Action Table
+///
+/// The table below indicates the behavior in each of the different
+/// cases.
+///
+///    - The "Value" column indicates the value being clamped.
+///    - The "Min" column indicates the minimum value of the range.
+///    - The "Max" column indicates the maximum value of the range.
+///    - The "Result" column indicates the value left on the top of the stack.
+///    - The "Success" column indicates whether the instruction succeeds, and if
+///      not what kind of error is returned:
+///       - ✅: success
+///       - ❗: recoverable error, with links to the error kind
+///       - ‼️: fatal error, with links to the error kind
+///    - The "Note" column briefly summarizes the action state in that case
+///
+/// | Value  | Min | Max | Result | Success | Note |
+/// | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
+/// | Less than Min | exists | exists | Min | ✅ | Value is replaced by Min |
+/// | Greater than Max | exists | exists | Max | ✅ | Value is replaced by Max |
+/// | In range [Min, Max] | exists | exists | Value | ✅ | Value remains unchanged |
+/// | missing | irrelevant | irrelevant | irrelevant| [❗…](crate::push_vm::stack::StackError::Underflow) | State is unchanged |
+///
+/// # Errors
+///
+/// If the stack access returns any error other than a
+/// [`StackError::Underflow`](crate::push_vm::stack::StackError::Underflow)
+/// then this returns that as a [`Error::Fatal`](crate::error::Error::Fatal)
+/// error.
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+pub struct Clamp;
```

Comment:
We should make this a `Clamp<T: Ord>{ _p : PhantomData<T> }` and then just use `Ord:.clamp`

Reply:
I didn't even realize there _was_ an `Ord::clamp` – thanks!

Reply:
We used `Ord::clamp` in the implementation but didn't yet make it generic. I'll do that in the equality and comparison pull request.